README.adoc

```diff
@@ -122,16 +122,24 @@ QueryResource.java
 include::finish/query/src/main/java/io/openliberty/guides/query/QueryResource.java[]
 ----
 
+The [hotspot=systemLoad file=0]`systemLoad` endpoint first gets all the hostnames by calling [hotspot=getSystems file=0]`getSystems()`.
+Then it loops through each hostname and calls [hotspot=getSystem file=0]`getSystem()` on each one. 
+This returns a `CompletionStage` interface which is processed asynchronously in the [hotspot=thenAcceptAsync file=0]`thenAcceptAsync()` method. 
+
 The `CompletionStage` interface represents a unit of computation.
 After a computation is complete, it can be either finished or chained with more `CompletionStage` interfaces using the [hotspot=thenAcceptAsync file=0]`thenAcceptAsync()` method. 
 Exceptions are handled in a callback provided to the [hotspot=exceptionally file=0]`exceptionally()` method, which behaves like a catch block.
 When you return a `CompletionStage` type in the resource, it doesn’t necessarily mean that the computation completed and the response was built.
 JAX-RS responds to the caller after the `CompletionStage` completes.
 
-A [hotspot=countdown1 hotspot=countdown2 hotspot=countdown3 file=0]`CountDownLatch` is used to track how many asynchronous requests that you're waiting on.
-By using the [hotspot=await file=0]`await` method, the program stops and waits until all our requests are complete. 
+In the [hotspot=systemLoad file=0]`systemLoad()` method a [hotspot=countdown1 hotspot=countdown2 hotspot=countdown3 file=0]`CountDownLatch` is used to keep track of the asynchronous requests. 
+The [hotspot=countdown2 hotspot=countdown3 file=0]`CountDownLatch` counts down whenever a request is complete.
+The `CountDownLatch` being at zero indicates that all requests are complete.
+By using the [hotspot=await file=0]`await()` method of the `CountDownLatch`, the program waits for all the requests to be complete before it continues.
```

Comment:
And, tie this section on CountDownLatch to before in the sense that it's used since we need to all async requests to be done before we can proceed?

Reply:
Updated to
```
By using the [hotspot=await file=0]`await()` method of the `CountDownLatch`, the program waits for all the asynchronous requests to be complete.
When all asynchronous requests are complete, the program will resume execution with all required data processed. 
```